### ISSUE TITLE
allow users to select their own seed value

### ISSUE THREAD

**Issue by frederic-mahe** [hildebra/Rarefaction#26]
Hi, replicability is important and it would great if `rtk` users could input their own pseudo-randomisation seed values. Doing so would allow different collaborators or different runs of the same pipeline to produce exactly the same rarefaction results, eliminating one source of variability.

I did not try it yet, but I assume that setting a literal value on the line 552 of `IO.cpp` would do the trick:
```C++
// IO.cpp:552
unsigned long long seed = (unsigned long long)chrono::high_resolution_clock::now().time_since_epoch().count();
```
A command line option such as `--seed INT` would be ideal.

**Comment by openpaul** [comment]
Thats a usefull feature request indeed. I added feature the feature in the branch https://github.com/hildebra/Rarefaction/tree/issue_26
For now its not tested widely and thus not merged with the master but feel free to build and use that branch.

You can provide the seed using `-s 42`.

**Comment by frederic-mahe** [comment]
thanks @openpaul it seems to work as expected:
```sh
cd ./Rarefaction/rtk/
git pull
git checkout remotes/origin/issue_26
make

# toy example
printf "OTUs\ts1\ts2\ts3\nA\t20\t20\t10\nB\t0\t2\t4\n" > tmp.table

# OTUs	s1	s2	s3
# A	20	20	10
# B	0	2	4


# no fix seed
for i in {1..10} ; do
    ./rtk memory -i tmp.table -d 10 -o mytmp_ -w 1 > /dev/null
    md5sum mytmp_rarefied_to_10.000000_n_0.tsv
done

# fix seed (md5 should be a7ce1a8c4962385b681b6648bc0206d1)
for i in {1..10} ; do
    ./rtk memory -i tmp.table -d 10 -o mytmp_ -w 1 -s 1 > /dev/null
    md5sum mytmp_rarefied_to_10.000000_n_0.tsv
done

# expected table:
# Rarefied	s1	s2	s3
# A	10	9	7
# B	0	1	3
```
Please feel free to close the issue.
